app/models.py

```diff
@@ -596,6 +605,56 @@ def add(self, domain_name, domain_type, soa_edit_api, domain_ns=[], domain_maste
             logging.debug(str(e))
             return {'status': 'error', 'msg': 'Cannot add this domain.'}
 
+    def create_reverse_domain(self, domain_name, domain_reverse_name):
+        """
+        Check the existing reverse lookup domain, 
+        if not exists create a new one automatically
+        """
+        domain_obj = Domain.query.filter(Domain.name == domain_name).first()
+        domain_auto_ptr = DomainSetting.query.filter(DomainSetting.domain == domain_obj).filter(DomainSetting.setting == 'auto_ptr').first()
+        domain_auto_ptr = strtobool(domain_auto_ptr.value) if domain_auto_ptr else False
+        system_auto_ptr = Setting.query.filter(Setting.name == 'auto_ptr').first()
+        system_auto_ptr = strtobool(system_auto_ptr.value)
+        self.name = domain_name
+        domain_id = self.get_id_by_name(domain_reverse_name)
+        if None == domain_id and \
+            (
+                system_auto_ptr or \
+                domain_auto_ptr
+            ):
+            result = self.add(domain_reverse_name, 'Master', 'INCEPTION-INCREMENT', '', '')
+            self.update()
+            if result['status'] == 'ok':
+                history = History(msg='Add reverse lookup domain %s' % domain_reverse_name, detail=str({'domain_type': 'Master', 'domain_master_ips': ''}), created_by='System')
+                history.add()
+            else:
+                return {'status': 'error', 'msg': 'Adding reverse lookup domain failed'}
+            domain_user_ids = self.get_user()
+            domain_users = []
+            u = User()
+            for uid in domain_user_ids:
+                u.id = uid
+                tmp = u.get_user_info_by_id()
+                domain_users.append(tmp.username)
+            if 0 != len(domain_users):
+                self.name = domain_reverse_name
+                self.grant_privielges(domain_users)
+                return {'status': 'ok', 'msg': 'New reverse lookup domain created with granted privilages'}
+            return {'status': 'ok', 'msg': 'New reverse lookup domain created without users'}
+        return {'status': 'ok', 'msg': 'Reverse lookup domain already exists'}
+
+    def get_reverse_domain_name(self, reverse_host_address):
+        if re.search('ip6.arpa', reverse_host_address):
+            for i in range(31,3,-1):
```

Comment:
In case of IPv4 - yes, same method
if you have host with 10.1.2.3 then you need to check existence of
2.1.10.in-addr.arpa first and then
1.10.in-addr.arpa
10.in-addr.arpa
and if not found then create "2.1.10.in-addr.arpa" zone and add "3" ptr record

For IPv6 i'm not sure what domain name you should create, maybe @jallakim can suggest